bookwyrm/settings.py

```diff
@@ -207,7 +207,10 @@
 REDIS_ACTIVITY_PORT = env("REDIS_ACTIVITY_PORT", 6379)
 REDIS_ACTIVITY_PASSWORD = env("REDIS_ACTIVITY_PASSWORD", None)
 REDIS_ACTIVITY_DB_INDEX = env("REDIS_ACTIVITY_DB_INDEX", 0)
-
+REDIS_ACTIVITY_URL = env(
+    "REDIS_ACTIVITY_URL",
+    f"redis://:{REDIS_ACTIVITY_PASSWORD}@{REDIS_ACTIVITY_HOST}:{REDIS_ACTIVITY_PORT}/{REDIS_ACTIVITY_DB_INDEX}",
+)
```

Comment:
Forgive my ignorance here - would it be possible to specify the protocol as its own .env variable, defaulting to `redis://` which could be changed to `unix://`, rather than making the entire url an env variable?

Reply:
I thought about that, but since these urls are constructed out of the other variables, it would be much more complex to assemble the urls for sockets as file paths. Also, this method theoretically would allow for redis over ssl (`rediss://`).
It basically allows for you to alternatively specify a full redis url instead of select parts.